tools/hdf_dump.py

```diff
@@ -145,8 +154,14 @@ def hdf_dump_from_dataset(dataset, hdf_dataset, parser_args):
     print("Labels for %s:" % data_key, labels[:3], "...", file=log.v5)
     max_label_len = max(map(len, labels))
     if data_key != default_data_input_key:
-      hdf_dataset['targets/labels'].create_dataset(data_key, (len(labels),), dtype="S%i" % (max_label_len + 1))
-      for i, label in enumerate(labels):
+      if data_key == default_data_target_key:
+        hdf_dataset['targets/labels'].create_dataset(valid_hdf_target_key,
+                                                     (len(labels),), dtype="S%i" % (max_label_len + 1))
+        for i, label in enumerate(labels):
+          hdf_dataset['targets/labels'][valid_hdf_target_key][i] =\
+            numpy.array(label, dtype="S%i" % (max_label_len + 1))
+      else:
+        hdf_dataset['targets/labels'].create_dataset(data_key, (len(labels),), dtype="S%i" % (max_label_len + 1))
         hdf_dataset['targets/labels'][data_key][i] = numpy.array(label, dtype="S%i" % (max_label_len + 1))
```

Comment:
This code line seems strange now. `i` is not defined here.

Reply:
(Btw, have you used PyCharm? I'm pretty sure it would warn about this.)

Reply:
Oh how could I miss this! Thanks!